pkg/scheduler/framework/plugins/README.md

```diff
@@ -68,21 +72,23 @@ your default values are valid. To do this, add something like this in
 // From here on, FooPluginArgs refers to the type defined in pkg/scheduler
 // definition, not the kube-scheduler definition since we're dealing with
 // post-default values.
-func ValidateFooPluginArgs(args config.FooPluginArgs) error {
+func ValidateFooPluginArgs(args config.FooPluginArgs) *field.Error {
 	if args.BarParam < 0 && args.BarParam > 100 {
-		return fmt.Errorf("not in valid range [0, 100]")
+		return field.Invalid(path.NewPath("barParam"), args.BarParam, "must be in the range [0, 100]")
 	}
+	return nil
 }
 ```
 
 Remember to write tests in `validation_pluginargs_test.go` to test your
-validator (again, tests will likely not pass until you do codegen).
+validator (again, tests will likely not pass until you run code generation).
 
 Next, add a test in `pkg/scheduler/apis/config/scheme/scheme_test.go` by
 setting `barParam`, which will allow you to test the whole pipeline.
 
-We have defined everything necessary to do codegen now. Remember to commit all
-your changes (not sure why this is needed) and do a `make clean` first. Then:
```

Comment:
Why `make clean` is required. I don't think a commit is required to run `./hack/update-codegen.sh`.

Reply:
Yeah, it's quite complicated. I still don't understand why certain things were failing for me even though I spent significant amount of time trying to debug hack scripts.

I know that verify scripts need a commit as they make a copy of your git head. I've just never had a situation when I had to commit before running code generation. 

I also think a curious reader of this guide might want to know why this is needed.

Reply:
Yeah, I also don't remember needing to commit. Last I tried, I think `make clean; make generated_files` was enough.